### ISSUE TITLE
ceph-volume report json: invalid character '-' in numeric literal

### ISSUE THREAD

**Issue by jshen28** [rook/rook#6824]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

With the following cluster definition, prepare osd job failed execution.

> cephcluster is run on a virtualized environment and virtio-blk device is used as metadata device

```yaml
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  creationTimestamp: "2020-12-14T02:26:30Z"
  finalizers:
  - cephcluster.ceph.rook.io
  generation: 2
  name: rook-ceph
  namespace: rook-ceph
  resourceVersion: "58742083"
  selfLink: /apis/ceph.rook.io/v1/namespaces/rook-ceph/cephclusters/rook-ceph
  uid: c67dfc92-3db3-11eb-a758-52540061a481
spec:
  cephVersion:
    image: registry-jinan-lab.inspurcloud.cn/library/ceph/ceph:14.2.15-li1
  cleanupPolicy:
    sanitizeDisks: {}
  crashCollector:
    disable: false
  dashboard:
    enabled: true
  dataDirHostPath: /var/lib/rook
  disruptionManagement: {}
  external:
    enable: false
  healthCheck:
    daemonHealth:
      mon: {}
      osd: {}
      status: {}
    livenessProbe:
      mon:
        disabled: true
      osd:
        disabled: true
  mgr: {}
  mon:
    count: 3
  monitoring: {}
  network:
    hostNetwork: true
    provider: ""
    selectors: null
  placement:
    mgr:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: node-role.ceph/mgr
              operator: In
              values:
              - enabled
      tolerations:
      - effect: NoSchedule
        key: pro
        operator: Equal
        value: master
    mon:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: node-role.ceph/mon
              operator: In
              values:
              - enabled
      tolerations:
      - effect: NoSchedule
        key: pro
        operator: Equal
        value: master
    osd:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: node-role.ceph/osd
              operator: In
              values:
              - enabled
      tolerations:
      - effect: NoSchedule
        key: pro
        operator: Equal
        value: master
  removeOSDsIfOutAndSafeToRemove: false
  resources:
    mgr:
      limits:
        cpu: "2"
        memory: 4Gi
      requests:
        cpu: 300m
        memory: 512Mi
    mon:
      limits:
        cpu: "2"
        memory: 24Gi
      requests:
        cpu: 300m
        memory: 512Mi
    osd:
      limits:
        cpu: "2"
        memory: 8Gi
      requests:
        cpu: 300m
        memory: 512Mi
  security:
    kms: {}
  storage:
    config:
      databaseSizeMB: ""
      metadataDevice: ""
      storeType: bluestore
      walSizeMB: ""
    nodes:
    - config: null
      devices:
      - config:
          metadataDevice: vdd
        name: vdc
      name: rookceph01
      resources: {}
    - config: null
      devices:
      - config:
          metadataDevice: vdd
        name: vdc
      name: rookceph02
      resources: {}
    - config: null
      devices:
      - config:
          metadataDevice: vdd
        name: vdc
      name: rookceph03
      resources: {}
    storageClassDeviceSets: null
    useAllDevices: false
status:
  ceph:
    capacity: {}
    details:
      TOO_FEW_OSDS:
        message: OSD count 0 < osd_pool_default_size 3
        severity: HEALTH_WARN
    health: HEALTH_WARN
    lastChecked: "2020-12-14T02:38:30Z"
  conditions:
  - lastHeartbeatTime: "2020-12-14T02:35:28Z"
    lastTransitionTime: "2020-12-14T02:26:37Z"
    message: Cluster progression is completed
    reason: ProgressingCompleted
    status: "False"
    type: Progressing
  - lastHeartbeatTime: "2020-12-14T02:28:21Z"
    lastTransitionTime: "2020-12-14T02:28:21Z"
    message: Failed to create cluster
    reason: ClusterFailure
    status: "True"
    type: Failure
  - lastHeartbeatTime: "2020-12-14T02:28:23Z"
    lastTransitionTime: "2020-12-14T02:28:23Z"
    message: Cluster created successfully
    reason: ClusterCreated
    status: "True"
    type: Ready
  - lastHeartbeatTime: "2020-12-14T02:32:26Z"
    lastTransitionTime: "2020-12-14T02:32:26Z"
    message: Cluster progression is completed
    reason: ProgressingCompleted
    status: "False"
  message: Cluster created successfully
  phase: Ready
  state: Created
  version:
    image: registry-jinan-lab.inspurcloud.cn/library/ceph/ceph:14.2.15-li1
    version: 14.2.15-4
```

```console
root@rookceph01:~/sjt/src# kubectl logs -n rook-ceph rook-ceph-osd-prepare-rookceph01-4dlp7 -p
2020-12-14 02:15:03.327819 I | cephcmd: desired devices to configure osds: [{Name:vdc OSDsPerDevice:1 MetadataDevice:vdd DatabaseSizeMB:0 DeviceClass:sjt IsFilter:false IsDevicePathFilter:false}]
2020-12-14 02:15:03.329587 I | rookcmd: starting Rook v1.5.1 with arguments '/rook/rook ceph osd provision'
2020-12-14 02:15:03.329608 I | rookcmd: flag values: --cluster-id=dd20bb53-3db1-11eb-a758-52540061a481, --data-device-filter=, --data-device-path-filter=, --data-devices=[{"id":"vdc","storeConfig":{"osdsPerDevice":1,"metadataDevice":"vdd","deviceClass":"sjt"}}], --drive-groups=, --encrypted-device=false, --force-format=false, --help=false, --location=, --log-flush-frequency=5s, --log-level=DEBUG, --metadata-device=, --node-name=rookceph01, --operator-image=, --osd-database-size=0, --osd-store=, --osd-wal-size=576, --osds-per-device=1, --pvc-backed-osd=false, --service-account=
2020-12-14 02:15:03.329613 I | op-mon: parsing mon endpoints: a=10.0.8.245:6789,b=10.0.8.246:6789,c=10.0.8.247:6789
2020-12-14 02:15:03.345623 I | op-osd: CRUSH location=root=default host=rookceph01
2020-12-14 02:15:03.348426 I | cephcmd: crush location of osd: root=default host=rookceph01
2020-12-14 02:15:03.348481 D | exec: Running command: nsenter --mount=/rootfs/proc/1/ns/mnt -- /usr/sbin/lvm --help
2020-12-14 02:15:03.351284 D | cephosd: failed to call nsenter. failed to execute nsenter. output: nsenter: failed to execute /usr/sbin/lvm: No such file or directory: exit status 1
2020-12-14 02:15:03.351421 D | cephosd: failed to lookup binary path "/rootfs/usr/sbin/lvm" on the host rootfs. stat /rootfs/usr/sbin/lvm: no such file or directory
2020-12-14 02:15:03.351517 D | exec: Running command: nsenter --mount=/rootfs/proc/1/ns/mnt -- /sbin/lvm --help
2020-12-14 02:15:03.362208 I | cephosd: successfully called nsenter
2020-12-14 02:15:03.362258 I | cephosd: binary "/sbin/lvm" found on the host, proceeding with osd preparation
2020-12-14 02:15:03.362268 D | exec: Running command: dmsetup version
2020-12-14 02:15:03.367473 I | cephosd: Library version:   1.02.170-RHEL7 (2020-03-24)
Driver version:    4.39.0
2020-12-14 02:15:03.525912 I | cephclient: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-12-14 02:15:03.526291 I | cephclient: generated admin config in /var/lib/rook/rook-ceph
2020-12-14 02:15:03.526622 D | cephosd: config file @ /etc/ceph/ceph.conf: [global]
fsid                      = 90ab0351-756a-4fcb-9b32-bfef8b32d6bc
mon initial members       = a b c
mon host                  = [v2:10.0.8.245:3300,v1:10.0.8.245:6789],[v2:10.0.8.246:3300,v1:10.0.8.246:6789],[v2:10.0.8.247:3300,v1:10.0.8.247:6789]
public addr               = 
cluster addr              = 
public network            = 10.0.8.0/24
cluster network           = 10.0.8.0/24
osd crush update on start = true
mon_max_pg_per_osd        = 300
osd pool default size     = 3
osd pool default min size = 2
osd pool default pg num   = 512
osd pool default pgp num  = 512

[osd]
run dir = /var/lib/rook

[mon]
run dir = /var/log/ceph

[mds]
run dir = /var/log/ceph

[mgr]
run dir = /var/log/ceph

[client.admin]
keyring = /var/lib/rook/rook-ceph/client.admin.keyring

2020-12-14 02:15:03.526671 I | cephosd: discovering hardware
2020-12-14 02:15:03.526692 D | exec: Running command: lsblk --all --noheadings --list --output KNAME
2020-12-14 02:15:03.532494 D | exec: Running command: lsblk /dev/loop1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.535502 W | inventory: skipping device "loop1". unsupported diskType loop
2020-12-14 02:15:03.535521 D | exec: Running command: lsblk /dev/vdd --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.538055 D | exec: Running command: sgdisk --print /dev/vdd
2020-12-14 02:15:03.590491 D | exec: Running command: udevadm info --query=property /dev/vdd
2020-12-14 02:15:03.594120 D | exec: Running command: lsblk --noheadings --pairs /dev/vdd
2020-12-14 02:15:03.601322 D | exec: Running command: lsblk /dev/loop6 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.605628 W | inventory: skipping device "loop6". exit status 1
2020-12-14 02:15:03.605652 D | exec: Running command: lsblk /dev/vdb --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.609534 D | exec: Running command: sgdisk --print /dev/vdb
2020-12-14 02:15:03.615861 D | exec: Running command: udevadm info --query=property /dev/vdb
2020-12-14 02:15:03.618166 D | exec: Running command: lsblk --noheadings --pairs /dev/vdb
2020-12-14 02:15:03.623446 I | inventory: skipping device "vdb" because it has child, considering the child instead.
2020-12-14 02:15:03.623512 D | exec: Running command: lsblk /dev/dm-1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.626025 D | exec: Running command: sgdisk --print /dev/dm-1
2020-12-14 02:15:03.628522 D | exec: Running command: udevadm info --query=property /dev/dm-1
2020-12-14 02:15:03.633249 D | exec: Running command: lsblk /dev/dm-0 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.636686 D | exec: Running command: sgdisk --print /dev/dm-0
2020-12-14 02:15:03.639107 D | exec: Running command: udevadm info --query=property /dev/dm-0
2020-12-14 02:15:03.641343 D | exec: Running command: lsblk /dev/loop4 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.643564 W | inventory: skipping device "loop4". unsupported diskType loop
2020-12-14 02:15:03.643583 D | exec: Running command: lsblk /dev/loop2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.645897 W | inventory: skipping device "loop2". unsupported diskType loop
2020-12-14 02:15:03.645918 D | exec: Running command: lsblk /dev/loop0 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.649558 W | inventory: skipping device "loop0". unsupported diskType loop
2020-12-14 02:15:03.649580 D | exec: Running command: lsblk /dev/loop7 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.651843 W | inventory: skipping device "loop7". exit status 1
2020-12-14 02:15:03.651866 D | exec: Running command: lsblk /dev/vdc --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.655311 D | exec: Running command: sgdisk --print /dev/vdc
2020-12-14 02:15:03.685199 D | exec: Running command: udevadm info --query=property /dev/vdc
2020-12-14 02:15:03.688136 D | exec: Running command: lsblk --noheadings --pairs /dev/vdc
2020-12-14 02:15:03.692823 D | exec: Running command: lsblk /dev/sda --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.696059 D | exec: Running command: sgdisk --print /dev/sda
2020-12-14 02:15:03.703334 D | exec: Running command: udevadm info --query=property /dev/sda
2020-12-14 02:15:03.706928 D | exec: Running command: lsblk --noheadings --pairs /dev/sda
2020-12-14 02:15:03.712819 D | exec: Running command: lsblk /dev/loop5 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.716341 W | inventory: skipping device "loop5". unsupported diskType loop
2020-12-14 02:15:03.716362 D | exec: Running command: lsblk /dev/vda --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.719835 D | exec: Running command: sgdisk --print /dev/vda
2020-12-14 02:15:03.723223 D | exec: Running command: udevadm info --query=property /dev/vda
2020-12-14 02:15:03.725370 D | exec: Running command: lsblk --noheadings --pairs /dev/vda
2020-12-14 02:15:03.730678 I | inventory: skipping device "vda" because it has child, considering the child instead.
2020-12-14 02:15:03.730748 D | exec: Running command: lsblk /dev/vda2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.733352 D | exec: Running command: udevadm info --query=property /dev/vda2
2020-12-14 02:15:03.735637 D | exec: Running command: lsblk /dev/vda1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.739198 D | exec: Running command: udevadm info --query=property /dev/vda1
2020-12-14 02:15:03.741330 D | exec: Running command: lsblk /dev/loop3 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.743982 W | inventory: skipping device "loop3". unsupported diskType loop
2020-12-14 02:15:03.744054 D | inventory: discovered disks are [0xc000159e60 0xc000117680 0xc0000c25a0 0xc0001178c0 0xc000117b00 0xc00015c360 0xc0001805a0]
2020-12-14 02:15:03.744083 I | cephosd: creating and starting the osds
2020-12-14 02:15:03.754992 D | cephosd: No Drive Groups configured.
2020-12-14 02:15:03.755383 D | cephosd: desiredDevices are [{Name:vdc OSDsPerDevice:1 MetadataDevice:vdd DatabaseSizeMB:0 DeviceClass:sjt IsFilter:false IsDevicePathFilter:false}]
2020-12-14 02:15:03.755452 D | cephosd: context.Devices are [0xc000159e60 0xc000117680 0xc0000c25a0 0xc0001178c0 0xc000117b00 0xc00015c360 0xc0001805a0]
2020-12-14 02:15:03.755493 D | exec: Running command: lsblk /dev/vdd --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:03.759687 D | exec: Running command: ceph-volume inventory --format json /dev/vdd
2020-12-14 02:15:08.752307 I | cephosd: device "vdd" is available.
2020-12-14 02:15:08.752368 I | cephosd: skipping device "vdd" that does not match the device filter/list ([{vdc 1 vdd 0 sjt false false}]). <nil>
2020-12-14 02:15:08.752381 I | cephosd: skipping 'dm' device "dm-1"
2020-12-14 02:15:08.752390 I | cephosd: skipping 'dm' device "dm-0"
2020-12-14 02:15:08.752404 D | exec: Running command: lsblk /dev/vdc --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:08.757490 D | exec: Running command: ceph-volume inventory --format json /dev/vdc
2020-12-14 02:15:13.906216 I | cephosd: device "vdc" is available.
2020-12-14 02:15:13.906253 I | cephosd: "vdc" found in the desired devices
2020-12-14 02:15:13.906259 I | cephosd: device "vdc" is selected by the device filter/name "vdc"
2020-12-14 02:15:13.906269 I | cephosd: skipping device "sda" because it contains a filesystem "iso9660"
2020-12-14 02:15:13.906274 I | cephosd: skipping device "vda2" because it contains a filesystem "ext4"
2020-12-14 02:15:13.906285 D | exec: Running command: udevadm info --query=property /dev/vda1
2020-12-14 02:15:13.909257 D | exec: Running command: lsblk /dev/vda1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME,KNAME
2020-12-14 02:15:13.912432 D | exec: Running command: ceph-volume inventory --format json /dev/vda1
2020-12-14 02:15:18.879184 I | cephosd: skipping device "vda1": ["Insufficient space (<5GB)"].
2020-12-14 02:15:18.879650 I | cephosd: configuring osd devices: {"Entries":{"vdc":{"Data":-1,"Metadata":null,"Config":{"Name":"vdc","OSDsPerDevice":1,"MetadataDevice":"vdd","DatabaseSizeMB":0,"DeviceClass":"sjt","IsFilter":false,"IsDevicePathFilter":false},"PersistentDevicePaths":["/dev/disk/by-path/pci-0000:00:09.0","/dev/disk/by-path/virtio-pci-0000:00:09.0"]}}}
2020-12-14 02:15:18.879736 I | cephclient: getting or creating ceph auth key "client.bootstrap-osd"
2020-12-14 02:15:18.880023 D | exec: Running command: ceph auth get-or-create-key client.bootstrap-osd mon allow profile bootstrap-osd --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/977445538
2020-12-14 02:15:19.437026 I | cephosd: configuring new device vdc
2020-12-14 02:15:19.437071 I | cephosd: using vdd as metadataDevice for device /dev/vdc and let ceph-volume lvm batch decide how to create volumes
2020-12-14 02:15:19.437096 D | exec: Running command: stdbuf -oL ceph-volume --log-path /tmp/ceph-log lvm batch --prepare --bluestore --yes --osds-per-device 1 --crush-device-class sjt /dev/vdc --db-devices /dev/vdd --report
2020-12-14 02:15:28.762581 D | exec: 
2020-12-14 02:15:28.762615 D | exec: Total OSDs: 1
2020-12-14 02:15:28.762619 D | exec: 
2020-12-14 02:15:28.762623 D | exec:   Type            Path                                                    LV Size         % of device
2020-12-14 02:15:28.762626 D | exec: ----------------------------------------------------------------------------------------------------
2020-12-14 02:15:28.762630 D | exec:   data            /dev/vdc                                                100.00 GB       100.00%
2020-12-14 02:15:28.762633 D | exec:   block_db        /dev/vdd                                                100.00 GB       100.00%
2020-12-14 02:15:28.762667 D | exec: --> passed data devices: 1 physical, 0 LVM
2020-12-14 02:15:28.762721 D | exec: --> relative data size: 1.0
2020-12-14 02:15:28.762729 D | exec: --> passed block_db devices: 1 physical, 0 LVM
2020-12-14 02:15:28.771476 D | exec: Running command: stdbuf -oL ceph-volume --log-path /tmp/ceph-log lvm batch --prepare --bluestore --yes --osds-per-device 1 --crush-device-class sjt /dev/vdc --db-devices /dev/vdd --report --format json
2020-12-14 02:15:37.640274 D | cephosd: ceph-volume report: --> passed data devices: 1 physical, 0 LVM
--> relative data size: 1.0
--> passed block_db devices: 1 physical, 0 LVM
[{"block_db": "/dev/vdd", "encryption": "None", "data": "/dev/vdc", "data_size": "100.00 GB", "block_db_size": "100.00 GB"}]
failed to configure devices: failed to initialize devices: failed to unmarshal ceph-volume report json: invalid character '-' in numeric literal
```

**Expected behavior:**

Prepare completed correctly.

**How to reproduce it (minimal and precise):**

Create a cluster and specify metadataDevice

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 18.04.3 LTS
* Kernel (e.g. `uname -a`):  5.0.0 5.0.0-29-generic-29-generic
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by jshen28** [comment]
By looking at the code [ceph-volume](https://github.com/ceph/ceph/blob/v14.2.15/src/ceph-volume/ceph_volume/terminal.py#L87), it will output something like `--> xxxx` to stderr, but [operator](https://github.com/rook/rook/blob/master/pkg/daemon/ceph/osd/volume.go#L610) will actually combine stderr output which breaks up json unmarshalling.

**Comment by jshen28** [comment]
Besides, why [cvReport](https://github.com/rook/rook/blob/master/pkg/daemon/ceph/osd/volume.go#L617) is not an array?

**Comment by jshen28** [comment]
By running command manually

```console
[root@rookceph01 /]# ceph-volume --log-path /tmp/ceph-log lvm batch --prepare --bluestore --yes --osds-per-device 1 /dev/vdc --db-devices /dev/vdd --report --format json
--> passed data devices: 1 physical, 0 LVM
--> relative data size: 1.0
--> passed block_db devices: 1 physical, 0 LVM
[{"block_db": "/dev/vdd", "encryption": "None", "data": "/dev/vdc", "data_size": "100.00 GB", "block_db_size": "100.00 GB"}]
[root@rookceph01 /]# ceph-volume --log-path /tmp/ceph-log lvm batch --prepare --bluestore --yes --osds-per-device 1 /dev/vdc --db-devices /dev/vdd                       
--> passed data devices: 1 physical, 0 LVM
--> relative data size: 1.0
--> passed block_db devices: 1 physical, 0 LVM
Running command: /usr/bin/ceph-authtool --gen-print-key
Running command: /usr/bin/ceph --cluster ceph --name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/ceph.keyring -i - osd new 5115f9dd-7f85-4504-a877-062fe3317c3b
Running command: /usr/sbin/vgcreate --force --yes ceph-18c4e1a5-dd1d-4b8c-9e38-bb301872bedc /dev/vdc
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
 stderr: WARNING: Failed to connect to lvmetad. Falling back to device scanning.
 stdout: Physical volume "/dev/vdc" successfully created.
 stdout: Volume group "ceph-18c4e1a5-dd1d-4b8c-9e38-bb301872bedc" successfully created
Running command: /usr/sbin/lvcreate --yes -l 25599 -n osd-block-5115f9dd-7f85-4504-a877-062fe3317c3b ceph-18c4e1a5-dd1d-4b8c-9e38-bb301872bedc
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
 stderr: WARNING: Failed to connect to lvmetad. Falling back to device scanning.
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, device-mapper library will manage device nodes in device directory.
 stdout: Logical volume "osd-block-5115f9dd-7f85-4504-a877-062fe3317c3b" created.
Running command: /usr/sbin/vgcreate --force --yes ceph-6eebe250-b867-4ef4-b742-7ab00940d588 /dev/vdd
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
 stderr: WARNING: Failed to connect to lvmetad. Falling back to device scanning.
 stdout: Physical volume "/dev/vdd" successfully created.
 stdout: Volume group "ceph-6eebe250-b867-4ef4-b742-7ab00940d588" successfully created
Running command: /usr/sbin/lvcreate --yes -l 25599 -n osd-db-ad6cfa06-ca9a-4343-a2f4-a87eed4aa730 ceph-6eebe250-b867-4ef4-b742-7ab00940d588
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
 stderr: WARNING: Failed to connect to lvmetad. Falling back to device scanning.
 stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, device-mapper library will manage device nodes in device directory.
 stdout: Logical volume "osd-db-ad6cfa06-ca9a-4343-a2f4-a87eed4aa730" created.
Running command: /usr/bin/ceph-authtool --gen-print-key
Running command: /usr/bin/mount -t tmpfs tmpfs /var/lib/ceph/osd/ceph-0
Running command: /usr/bin/chown -h ceph:ceph /dev/ceph-18c4e1a5-dd1d-4b8c-9e38-bb301872bedc/osd-block-5115f9dd-7f85-4504-a877-062fe3317c3b
Running command: /usr/bin/chown -R ceph:ceph /dev/mapper/ceph--18c4e1a5--dd1d--4b8c--9e38--bb301872bedc-osd--block--5115f9dd--7f85--4504--a877--062fe3317c3b
Running command: /usr/bin/ln -s /dev/ceph-18c4e1a5-dd1d-4b8c-9e38-bb301872bedc/osd-block-5115f9dd-7f85-4504-a877-062fe3317c3b /var/lib/ceph/osd/ceph-0/block
Running command: /usr/bin/ceph --cluster ceph --name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/ceph.keyring mon getmap -o /var/lib/ceph/osd/ceph-0/activate.monmap
 stderr: got monmap epoch 3
Running command: /usr/bin/ceph-authtool /var/lib/ceph/osd/ceph-0/keyring --create-keyring --name osd.0 --add-key AQAdLddf8hDNABAAgozkCcFuuMiqSTKv5N7fVw==
 stdout: creating /var/lib/ceph/osd/ceph-0/keyring
added entity osd.0 auth(key=AQAdLddf8hDNABAAgozkCcFuuMiqSTKv5N7fVw==)
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/keyring
Running command: /usr/bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0/
Running command: /usr/bin/chown -h ceph:ceph /dev/ceph-6eebe250-b867-4ef4-b742-7ab00940d588/osd-db-ad6cfa06-ca9a-4343-a2f4-a87eed4aa730
Running command: /usr/bin/chown -R ceph:ceph /dev/mapper/ceph--6eebe250--b867--4ef4--b742--7ab00940d588-osd--db--ad6cfa06--ca9a--4343--a2f4--a87eed4aa730
Running command: /usr/bin/ceph-osd --cluster ceph --osd-objectstore bluestore --mkfs -i 0 --monmap /var/lib/ceph/osd/ceph-0/activate.monmap --keyfile - --bluestore-block-db-path /dev/ceph-6eebe250-b867-4ef4-b742-7ab00940d588/osd-db-ad6cfa06-ca9a-4343-a2f4-a87eed4aa730 --osd-data /var/lib/ceph/osd/ceph-0/ --osd-uuid 5115f9dd-7f85-4504-a877-062fe3317c3b --setuser ceph --setgroup ceph
 stderr: parse error setting 'cluster_addr' to ''
 stderr: parse error setting 'public_addr' to ''
 stderr: 2020-12-14 09:15:30.270 7f035ed91a80 -1 asok(0x55c85fa26000) AdminSocketConfigObs::init: failed: AdminSocket::bind_and_listen: failed to bind the UNIX domain socket to '/var/lib/rook/ceph-osd.0.asok': (98) Address already in use
 stderr: 2020-12-14 09:15:30.278 7f035ed91a80 -1 bluestore(/var/lib/ceph/osd/ceph-0/) _read_fsid unparsable uuid
--> ceph-volume lvm prepare successful for: /dev/vdc
```

Looks like even if both osd device and metadata device are hdd (virtio), lvm batch still manages to create osd and db separately,

```console
[root@rookceph01 /]# pvs
  Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
  Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
  WARNING: Failed to connect to lvmetad. Falling back to device scanning.
  PV         VG                                        Fmt  Attr PSize    PFree   
  /dev/vdb   vg-data                                   lvm2 a--  <300.00g 1020.00m
  /dev/vdc   ceph-18c4e1a5-dd1d-4b8c-9e38-bb301872bedc lvm2 a--  <100.00g       0 
  /dev/vdd   ceph-6eebe250-b867-4ef4-b742-7ab00940d588 lvm2 a--  <100.00g       0 
[root@rookceph01 /]# lvs
  Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
  Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
  WARNING: Failed to connect to lvmetad. Falling back to device scanning.
  LV                                             VG                                        Attr       LSize    Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  osd-block-5115f9dd-7f85-4504-a877-062fe3317c3b ceph-18c4e1a5-dd1d-4b8c-9e38-bb301872bedc -wi-a----- <100.00g                                                    
  osd-db-ad6cfa06-ca9a-4343-a2f4-a87eed4aa730    ceph-6eebe250-b867-4ef4-b742-7ab00940d588 -wi-a----- <100.00g                                                    
  lv-docker                                      vg-data                                   -wi-ao----  150.00g                                                    
  lv-kubelet                                     vg-data                                   -wi-ao----  149.00g             
```

And osd is registered,

```console
[root@rookceph01 /]# ceph osd tree
ID CLASS WEIGHT TYPE NAME    STATUS REWEIGHT PRI-AFF 
-1            0 root default                         
 0            0 osd.0          down        0 1.00000 
```

**Comment by jshen28** [comment]
OK, seems 14.2.15 brings some breaking changes......

**Comment by travisn** [comment]
In v14.2.13 and v14.2.14 there was an issue with ceph-volume that I thought would be fixed in v14.2.15. I'm running the tests now against v14.2.15 with #6827 to confirm. The original issue was #6532.

**Comment by jshen28** [comment]
I think it's working if i do not specify metadata devices.
